File: fibonacci.js
/**
 * Created by elycruz on 2/4/15.
 * Returns the Fibonacci Series up to `limit` as an array.
 * @param limit {Number}
 * @returns {Array}
 */
function fib (limit) {
    limit = limit || 1000;
    var a = 0,
        b = 1,
        out = [];
    while (a <= limit) {
        out.push(a);
        out.push(b);
        a = a + b;
        b = a + b;
    }
    out.pop();
    return out;
}

console.log( fib(1000).join(', ') );

File: fibonacci.rb
class Fibonacci
  def initialize
    @array = [1, 1]
  end

  def sum
    while @array[-1] < 1000 do
      @array << @array[-2] + @array[-1]
    end
    @array.delete_at(-1)
  end

  def print
    p @array.join(' ')
  end
end

f = Fibonacci.new()
f.sum
f.print
